controller/src/main/java/cern/jarrace/controller/rest/controller/AgentContainerController.java

```diff
@@ -0,0 +1,104 @@
+package cern.jarrace.controller.rest.controller;
+
+import cern.jarrace.controller.domain.Entrypoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * @author timartin
+ * Controller that exposes services to manage {@link Entrypoint}s
+ */
+@RestController
+@RequestMapping("/jarrace")
+public class AgentContainerController {
```

Comment:
ANother note abour searating concerns: I think we are have an additional type of mixing concerns. I think we have three differnt types of clients. I think each of them should have their different controller. No need for them to see the other functionality:
* THe gradle task: Basically only needs to deploy
* The (later) running and debugging gui/client ... needs all the methods to start/stop/debug an agent/entrypoint
* The agents themselves: They need to register whatever entrypoints they have